File: Problems/Algorithms/1491. Average Salary Excluding the Minimum and Maximum Salary/AverageSalary.java
class Solution {
    public double average(int[] salary) {
        int minSalary = Integer.MAX_VALUE;
        int maxSalary = Integer.MIN_VALUE;
        int sum = 0;

        for (int curr : salary) {
            sum += curr;
            minSalary = Math.min(minSalary, curr);
            maxSalary = Math.max(maxSalary, curr);
        }

        return (double)(sum - minSalary - maxSalary) / (double)(salary.length - 2);
    }
}

File: Problems/Algorithms/1491. Average Salary Excluding the Minimum and Maximum Salary/AverageSalary.kt
class Solution {
    fun average(salary: IntArray): Double {
        var sum = 0
        var min = minOf(salary[0], salary[1])
        var max = maxOf(salary[0], salary[1])
        for (i in 2..salary.size-1) {
            val curr = salary[i]
            if (min > curr) {
                sum += min
                min = curr
            } else if (max < curr) {
                sum += max
                max = curr 
            } else {
                sum += curr
            }
        }
        
        return sum / (salary.size-2).toDouble()
    }
}

